mesonbuild/dependencies/base.py

```diff
@@ -345,6 +345,24 @@ def generate_link_whole_dependency(self) -> Dependency:
         new_dep.libraries = []
         return new_dep
 
+    def get_as_static(self, recursive: bool) -> Dependency:
+        from ..build import BothLibraries
+
+        new_dep = copy.copy(self)
+        new_dep.libraries = [lib.static if isinstance(lib, BothLibraries) else lib for lib in self.libraries]
```

Comment:
Maybe to avoid the need for the `isinstance` check here it would be better to add a `as_static` and `as_shared` method directly to the target types so that you can just `new_dep.libraries = [l.as_static() for l in self.libraries]`?